### PR TITLE
Migrate trunc as LLK op

### PR DESCRIPTION
**Pull request by Aswinmcw**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/21956

### Problem description
trunc op implemented as composite op

### What's changed
Migrate it as LLK op

DEVICE KERNEL DURATION [ns]
Screenshot 2025-06-09 at 5 03 04 PM

Edge case behaviour:
| Input | DataType | Torch output | TT output     |
|-------|----------|--------------|---------------|
| inf   | bfloat16 | inf          | inf           |
| -inf  | bfloat16 | -inf         | -inf          |
| nan   | bfloat16 | nan          | inf           |
| inf   | float32  | inf          | 2147483648    |
| -inf  | float32  | -inf         | -2147483648   |
| nan   | float32  | nan          | 2147483648    |

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/15673911553) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/runs/15673915727) CI with demo tests passes (if applicable)
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [x] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/eltwise/test_composite.py [hunk 1]
```diff
@@ -604,45 +604,6 @@ def test_unary_composite_triu_ttnn(input_shapes, device):
     assert comp_pass
```
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/eltwise/test_composite.py:568]: we can have atleast one variant of these PCC tests. we don't have to remove all of them yet. 
but pls test passing seed value to data_gen_with_range

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/eltwise/test_unary.py:1]: 1. Add test with edge-case coverage 
2. use linear dataset that covers the entire range of 2**16 (65K) - this can be a follow-up PR 
3. record inf/nan behaviour - for supported dtype

**Review by KalaivaniMCW — APPROVED** [review]
LGTM ! Thanks for updating the edge case behaviour. We will revisit this along with #22794 to determine the behaviour and how to achieve it without affecting perf too much.

**Review by nvelickovicTT — APPROVED** [review]

**Review by VirdhatchaniKN — APPROVED** [review]

**Review by nvelickovicTT — APPROVED** [review]

**Review by patrickroberts — APPROVED** [review]

**Review by dchenTT — APPROVED** [review]
